platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/StopWatch.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple stop watch that can be used to measure elapsed time.
+ */
+public class StopWatch {
+    private long startTime;
+    private long endTime;
+    private boolean running;
+    private boolean suspended;
+
+    /**
+     * Creates a new StopWatch.
+     */
+    public StopWatch() {
+        this.running = false;
+        this.suspended = false;
+    }
+
+    /**
+     * Starts StopWatch
+     */
+    public void start() {
+        if (running) {
+            throw new IllegalStateException("StopWatch is already running.");
+        }
+        this.startTime = System.nanoTime();
+        this.running = true;
+    }
+
+    /**
+     * Stops StopWatch only after being running
+     */
+    public void stop() {
+        if (!running || suspended) {
+            throw new IllegalStateException("StopWatch is not running.");
+        }
+        this.endTime = System.nanoTime();
+        this.running = false;
+    }
+
+    /**
+     * Suspends StopWatch only after being started
+     */
+    public void suspend() {
+        if (!running || suspended) {
+            throw new IllegalStateException("StopWatch is not running or is already suspended.");
+        }
+        this.endTime = System.nanoTime();
+        this.suspended = true;
+    }
+
+    /**
+     * Resumes StopWatch only after being suspended
+     */
+    public void resume() {
+        if (!suspended) {
+            throw new IllegalStateException("StopWatch is not suspended.");
+        }
+        this.startTime += (System.nanoTime() - endTime); // Adjusting the startTime
```

Comment:
I ask myself if it is correct to change the startTime in that case or if we need to track the time the watch has been suspended and do the calculation when someone needs the elapsed time